### PR TITLE
Add pydantic

### PR DESCRIPTION
**Pull request by drew2a**
Tribler config is a pydantic's `BaseModel` now.

It still writes and loads data from `ini`-file (named somewhy `triblerd.conf` in Tribler)

More information: https://pydantic-docs.helpmanual.io/usage/models/

### CODE REVIEW

File: src/tribler-core/tribler_core/config/tribler_config_section.py [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+            return None
+
+        path = Path(value)
+        if path.is_absolute():
+            return path
+
+        return state_dir / path
+
+    @root_validator(pre=True)
+    def convert_from_none_string_to_none_type(cls, values):
```
Comment (ichorid) [inline src/tribler-core/tribler_core/config/tribler_config_section.py:42]: :+1:

File: src/tribler-core/tribler_core/modules/libtorrent/settings.py [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+
+    anon_proxy_server_ip: str = '127.0.0.1'
+    anon_proxy_server_ports: List[str] = ['-1', '-1', '-1', '-1', '-1']
+    anon_proxy_auth: str = ''
+
+    _port_validator = validator('port', 'anon_listen_port', allow_reuse=True)(validate_port_with_minus_one)
+
+    @validator('anon_proxy_server_ip')
+    def validate_ip_address(cls, v):
+        pattern = r'^((25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])\.){3}(25[0-5]|2[0-4][0-9]|1[0-9][0-9]|[1-9]?[0-9])$'
```
Comment (ichorid) [inline src/tribler-core/tribler_core/modules/libtorrent/settings.py:40]: :sunglasses:

Comment (qstokkink) [reply src/tribler-core/tribler_core/modules/libtorrent/settings.py:40]: I'd suggest replacing this method with:

```python
def validate_ip_address(cls, v):
    return v if ipaddress.IPv4Network(v) else None
```

This is slightly easier to read and also gives in-depth tracebacks:

```python
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
  File "E:\Python38\lib\ipaddress.py", line 1449, in __init__
    self.network_address = IPv4Address(addr)
  File "E:\Python38\lib\ipaddress.py", line 1252, in __init__
    self._ip = self._ip_int_from_string(addr_str)
  File "E:\Python38\lib\ipaddress.py", line 1149, in _ip_int_from_string
    raise AddressValueError("%s in %r" % (exc, ip_str)) from None
ipaddress.AddressValueError: Octet 256 (> 255) not permitted in '256.1.0.2'
```

**Comment by drew2a** [comment]
retest it please

**Comment by drew2a** [comment]
retest this please

File: src/tribler-core/tribler_core/modules/libtorrent/settings.py (hunk 2 above)
Comment (drew2a) [reply src/tribler-core/tribler_core/modules/libtorrent/settings.py]: Replaced, thanks.

**Comment by drew2a** [comment]
retest this please

**Review by ichorid — APPROVED** [review]

**Comment by ichorid** [comment]
:tada: :clap:

**Pull request merged**